### PR TITLE
Update to 1.59.0-C-5

### PR DESCRIPTION
**Pull request by jbradleysmith**
Overview:
build ionic.efirom using CONFIG=ionic_efirom
hii strings are optionally include with CONFIG=ionic_efirom support for ionicvf device 1dd8:1003
support for 9k MTU

Detailed internal patch list:
- build pensando images
* config: add local pen_ionic_efirom config for PEN_IONIC_EFIROM
* ionic: ifdef PEN_IONIC_EFIROM for oprom bar test
* add in hii_strings files with whitespace=fix
* conditionally include hii based on PEN_IONIC_EFIROM
* ionic: check for rom bar on PF only
* ionic: bump ipxe DRV_VERSION to 2.0
* efi_hii restore whitespace to upstream
- Double commit of 9kMTU support and AMD pensando FW version print
* Re-design the code flow to support 9K MTU and print AMD-Pensando FW version
* Dump the FW Version in iPXE console if logging enable
- ionic: use dma api to map buffers before passing to device (#19)
- ionic_vf: skip hii commands for VFs

### CODE REVIEW

File: src/drivers/net/ionic.c [hunk 1]
```diff
@@ -16,10 +16,12 @@
  *
  */
 
+#include "ipxe/iobuf.h"
 FILE_LICENCE(GPL2_OR_LATER_OR_UBDL);
 
 #include "ionic.h"
 
+
```
Comment (emusln) [inline src/drivers/net/ionic.c:24]: Unnecessary blank line?

File: src/drivers/net/ionic.c [hunk 2]
```diff
@@ -153,8 +156,31 @@ int ionic_start_queues(struct ionic *ionic)
...
+err_tx_qcq_enable:
+	ionic_qcq_disable(ionic->lif->txqcqs);
+	return err;
 }
 
+static void ionic_lif_rxtx_deinit(struct ionic *ionic)
+{
+	if (ionic->qs_running) {
+		DBG_OPROM_ERR(ionic, "queues are not stopped - "
+                      "abort ionic_lif_rxtx_deinit\n");
```
Comment (emusln) [inline src/drivers/net/ionic.c:168]: Why not stop the queues here if they are running?

File: src/drivers/net/ionic.c [hunk 3]
```diff
@@ -167,21 +193,36 @@ static int ionic_open(struct net_device *netdev)
...
+		goto err_start_device;
 	}
 
-	ionic_check_link(netdev);
+	if (ionic_is_link_up(ionic)) {
+		err = ionic_start_queues(ionic);
+		if (err)
+			goto err_lif_rxtx_init;
+	} else {
+		DBG_OPROM_INFO(ionic, "Link is down-skipping ionic_start_queues\n");
```
Comment (emusln) [inline src/drivers/net/ionic.c:214]: I read this wrong at first and was wondering what the work "down-skipping" meant.  Putting spaces around the hyphen would help clear this up.

File: src/drivers/net/ionic.c [hunk 4]
```diff
@@ -235,11 +270,17 @@ static void ionic_close(struct net_device *netdev)
...
+		"rx_alloc_fail_cnt: %lld rx_desc_avail: %d\n",
 		stats->rx_done.cnt, stats->rx_doorbell.cnt,
 		stats->rx_alloc_fail.cnt, ionic_q_space_avail(rxq));
+	DBG_OPROM_INFO(ionic, "MTU size :%ld\n", netdev->mtu);
 
-	ionic_stop_queues(ionic);
+	if (ionic_is_link_up(ionic)) {
+		ionic_stop_queues(ionic);
+	} else {
+		DBG_OPROM_INFO(ionic, "Link is down-skipping ionic_stop_queues\n");
```
Comment (emusln) [inline src/drivers/net/ionic.c:281]: Spaces around hyphen

File: src/drivers/net/ionic.c [hunk 5]
```diff
@@ -527,31 +567,64 @@ void ionic_handle_fw_down(struct ionic *ionic)
...
+	}
+	if (netdev_is_open(ionic->netdev)) {
+		ionic_lif_rxtx_deinit(ionic);
+	}
 
 	// Stop the device
 	ionic_stop_device(ionic);
 }
 
+
```
Comment (emusln) [inline src/drivers/net/ionic.c:584]: Unnecesary blank line

File: src/drivers/net/ionic.c [hunk 6]
```diff
@@ -527,31 +567,64 @@ void ionic_handle_fw_down(struct ionic *ionic)
...
+		if (ionic_is_link_up(ionic)) {
+			err = ionic_start_queues(ionic);
+			if (err) {
+				DBG_OPROM_ERR(ionic, "Failed in ionic_start_queue :%d\n",
+				  err);
+				goto err_lif_rxtx_init;
+			}
+		} else {
+			DBG_OPROM_INFO(ionic, "Link is down-skipping ionic_start_queues\n");
```
Comment (emusln) [inline src/drivers/net/ionic.c:617]: Spaces around hyphen

**Review by emusln — COMMENTED** [review]
I don't think any of my comments are more than cosmetic and could be ignored, but you might look into them anyway.

**Comment by jbradleysmith** [comment]
This PR brings the public repo up to the ipxe-dev repo.  These comments should probably be addressed in the ipxe-dev repo and then get merged over here.  This PR matches what we are shipping to Oracle.

**Comment by emusln** [comment]
> This PR brings the public repo up to the ipxe-dev repo. These comments should probably be addressed in the ipxe-dev repo and then get merged over here. This PR matches what we are shipping to Oracle.

I kinda thought so.  Works for me.

**Review by emusln — APPROVED** [review]

**Pull request merged**